### ISSUE TITLE
Disabled C++ test batchnorm_test.cc - TestIterAll

### ISSUE THREAD

**Issue by marcoabreu** [apache/incubator-mxnet#10131]
https://github.com/apache/incubator-mxnet/blame/master/tests/cpp/operator/batchnorm_test.cc#L1072

Test has been disabled as part of MKLDNN

**Comment by zheng-da** [comment]
To make it clear, it's disabled not because of MKLDNN. It's disabled because the unit test has to be changed for the new interface used by batchnorm. I don't think you should label it as MKL.

**Comment by cjolivier01** [comment]
I believe TestIterAll was disabled because it was mysteriously failing.

**Comment by zheng-da** [comment]
is it possible that the problem was caused by the potential memory error I found in the unit test?

**Comment by zheng-da** [comment]
@cjolivier01 have you tried any memory debug tools such as valgrind and address sanitizer?

**Comment by cjolivier01** [comment]
i don’t think it was caused by that because that codepwth was designed
specifically for batch norm test, so i don’t believe any memory problems
occurred
